python/mxnet/numpy/multiarray.py

```diff
@@ -1993,10 +1993,12 @@ def array(object, dtype=None, ctx=None):
     """
     if ctx is None:
         ctx = current_context()
-    if isinstance(object, ndarray):
+    if isinstance(object, (ndarray, _np.ndarray)):
         dtype = object.dtype if dtype is None else dtype
     else:
         dtype = _np.float32 if dtype is None else dtype
+        if hasattr(object, "dtype"):
+            dtype = object.dtype
         if not isinstance(object, (ndarray, _np.ndarray)):
```

Comment:
I feel that this if-statement is not necessary because we've tested it in Line 1996.

Reply:
removed